tests/cmake.py

```diff
@@ -0,0 +1,102 @@
+import os
+import sys
+import subprocess
+
+
+class CMake:
+    def __init__(self, factory):
+        self.runs = dict()
+        self.factory = factory
+
+    def compile(self, targets, options=None):
+        key = (
+            ";".join(targets),
+            ";".join(["{}={}".format(k, options[k]) for k in options]),
```

Comment:
Pretty sure `options` can be `None` here and that can not be iteratred.

Reply:
I think `options` is meant to be a `**kwargs` passed through, so is a dictionary.  Much better to write `";".join("{}={}".format(k, v) for k, v in options.items())`.  (Also note you don't need to wrap into an extra list, you can directly use a generator expression instead of a list comprehension first.  Further you could try and use f-strings - but I have no idea if we depends on recent enough python versions for that here.  It might be worth it)